test/e2e/e2e_test.go

```diff
@@ -1252,6 +1253,185 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
+		framework.ExpectNoError(err)
+		if len(nodes.Items) < 3 {
+			framework.Failf(
+				"Test requires >= 3 Ready nodes, but there are only %v nodes",
+				len(nodes.Items))
+		}
+		ips := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
+		worker1NodeInfo = nodeInfo{
+			name:   nodes.Items[1].Name,
+			nodeIP: ips[1],
+		}
+		worker2NodeInfo = nodeInfo{
+			name:   nodes.Items[2].Name,
+			nodeIP: ips[2],
+		}
+	})
+
+	ginkgo.AfterEach(func() {
+		// tear down the containers simulating the gateways
+		if cid, _ := runCommand("docker", "ps", "-qaf", fmt.Sprintf("name=%s", gwContainerName)); cid != "" {
+			if _, err := runCommand("docker", "rm", "-f", gwContainerName); err != nil {
+				framework.Logf("failed to delete the gateway test container %s %v", gwContainerName, err)
+			}
+		}
+	})
+
+	ginkgo.It("Should validate connectivity to an external gateway's loopback address via a pod with external gateway annotations enabled", func() {
+
+		var (
+			pingSrc           string
+			validIP           net.IP
+			exGWRemoteCidr    = fmt.Sprintf("%s/32", exGWRemoteIp)
+			command           = []string{"bash", "-c", "sleep 20000"}
+			frameworkNsFlag   = fmt.Sprintf("--namespace=%s", f.Namespace.Name)
+			testContainer     = fmt.Sprintf("%s-container", srcPingPodName)
+			testContainerFlag = fmt.Sprintf("--container=%s", testContainer)
+		)
+
+		// start the container that will act as an external gateway
+		_, err := runCommand("docker", "run", "-itd", "--privileged", "--network", ciNetworkName, "--name", gwContainerName, "centos")
+		if err != nil {
+			framework.Failf("failed to start external gateway test container %s: %v", gwContainerName, err)
+		}
+		// retrieve the container ip of the external gateway container
+		exGWIp, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, gwContainerName)
+		if err != nil {
+			framework.Failf("failed to start external gateway test container: %v", err)
+		}
+		// trim newline from the inspect output
+		exGWIp = strings.TrimSuffix(exGWIp, "\n")
+		if ip := net.ParseIP(exGWIp); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", gwContainerName, exGWIp)
+		}
+
+		// create the pod that acts as a proxy for egress traffic to the external gateway
+		createGenericPod(f, gatewayPodName, worker1NodeInfo.name, defaultNamespace, command)
+		// wait for pod setup to return a valid address
+		err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+			pingSrc, err = getPodAddress(gatewayPodName, defaultNamespace)
+			if err != nil {
+				framework.Logf("retrying ... error trying to get pod  %s address: %v", srcPingPodName, err)
+				return false, nil
+			}
+			validIP = net.ParseIP(pingSrc)
+			if validIP == nil {
+				return false, nil
+			}
+			return true, nil
+		})
+
+		// add the annotations to the pod to enable the gateway forwarding
+		annotateArgs := []string{
```

Comment:
maybe just add a comment here saying that we have to fake out the multus annotation so that the pod IP is really an IP of an external container for testing purposes